util/src/main/java/org/datacommons/util/StringUtil.java

```diff
@@ -106,7 +115,7 @@ public static boolean isBool(String val) {
   }
 
   public static String getValidISO8601DatePattern(String dateValue) {
-    for (String pattern : DATE_PATTERNS) {
+    for (String pattern : DATE_PATTERNS.keySet()) {
```

Comment:
Curious why not optimize this function? It gets called from mcf checker in the fast path too?

Reply:
You are correct; I missed this path. I will apply the same optimization and also re-run the performance tests.

Reply:
Done.